core/templates/dev/head/forms/formBuilder.js

```diff
@@ -598,9 +603,27 @@ oppia.factory('rteHelperService', [
       // Returns a DOM node.
       createRteElement: function(componentDefn, customizationArgsDict) {
         var el = $('<img/>');
-        el.attr('src', componentDefn.iconDataUrl);
+        if (componentDefn.previewUrlTemplate) {
+          var interpolatedUrl = $interpolate(
+            componentDefn.previewUrlTemplate, false, null, true)(
+              angular.extend(customizationArgsDict,
```

Comment:
Can we only do the extending only if explorationContextService.getPageContext() is editor/learner? (Maybe the cleanest way to do that is to add a boolean isInExplorationContext() method to the explorationContextService.)